### PR TITLE
mpdecimal 2.5.0 (new formula)

### PR DESCRIPTION
**Pull request by lpinca**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Previous attempts in chronological order:

- https://github.com/Homebrew/legacy-homebrew/pull/46545
- https://github.com/Homebrew/legacy-homebrew/pull/48557
- https://github.com/Homebrew/legacy-homebrew/pull/49756
- https://github.com/Homebrew/homebrew-core/pull/36417

### CODE REVIEW

File: Formula/libmpdec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  def install
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    (testpath/"test.c").write <<~EOS
+      #include <mpdecimal.h>
+      #include <string.h>
```
Comment (carlocab) [inline Formula/libmpdec.rb:17]: I think you also need an `#include <assert.h>` here.

Comment (lpinca) [reply Formula/libmpdec.rb:18]: Yes, copy/paste error. Thank you.

**Comment by carlocab** [comment]
This library already looks widely distributed: https://repology.org/project/mpdecimal/versions

However, everyone else seems to call it `mpdecimal`. I think we should follow suit.

Other than the name, this LGTM.

File: Formula/libmpdec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+class Libmpdec < Formula
```
Comment (carlocab) [inline Formula/libmpdec.rb:1]:
```suggestion
class Mpdecimal < Formula
```

**Review by carlocab — CHANGES_REQUESTED** [review]
Just one more thing and this is 👍 to me.

File: Formula/libmpdec.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/libmpdec.rb:1]: That means you'll also need to change the filename

**Comment by lpinca** [comment]
@carlocab @SMillerDev I would prefer to keep the `libmpdec` for consistency with https://packages.ubuntu.com/focal/libdevel/libmpdec-dev https://packages.debian.org/buster/libmpdec-dev

**Comment by carlocab** [comment]
They call the source package `mpdecimal`, though: https://packages.ubuntu.com/source/groovy/mpdecimal

We can always add a `libmpdec` alias.

**Comment by lpinca** [comment]
Ok, done.

**Comment by carlocab** [comment]
Great. In the `Aliases` directory, you can add an alias
```
libmpdec -> ../Formula/mpdecimal.rb
```
so that you can also do `brew install libmpdec`, etc.

**Comment by carlocab** [comment]
Ah. `Aliases/libmpdc` needs to be a symlink. Just do
```
cd $(brew --repo homebrew/core)/Aliases
ln -s ../Formula/mpdecimal.rb libmpdec
```

**Comment by lpinca** [comment]
Yes, fighting with Windows :) I'm not used to work on Windows.

**Comment by carlocab** [comment]
If it's too much trouble, I can also push a commit that adds the symlink. But you have to remove the original non-symlink alias first.

**Comment by lpinca** [comment]
Ok, it should be good now.

**Review by carlocab — APPROVED** [review]
Thanks, @lpinca! Nice to see this one finally get over the line.

**Pull request closed without merging**